### PR TITLE
uscorer4 procedure transformer

### PR DESCRIPTION
**Pull request by ilaflamme-va**

### CODE REVIEW

File: data-query/pom.xml [hunk 1]
```diff
@@ -12,7 +12,7 @@
   <packaging>jar</packaging>
   <properties>
     <fall-risk.version>1.0.29</fall-risk.version>
-    <fhir-resources.version>3.0.5</fhir-resources.version>
+    <fhir-resources.version>3.0.7-SNAPSHOT</fhir-resources.version>
```
Comment (taylorrassmannlibertyits) [inline data-query/pom.xml:15]: Remove SNAPSHOT once procedure model is merged

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/R4Transformers.java [hunk 2]
```diff
@@ -16,6 +18,29 @@
...
+    }
+    return CodeableConcept.builder().coding(List.of(fhirCoding)).build();
+  }
+
+  /**
+   * Convert the optional coding to a FHIR coding and wrap it in a codeable concept. Returns null if
+   * it cannot be converted.
+   */
+  public static CodeableConcept asCodeableConceptWrapping(Optional<DatamartCoding> coding) {
+    if (coding == null || coding.isEmpty()) {
```
Comment (taylorrassmannlibertyits) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/R4Transformers.java:38]: You can just use isEmpty here.

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/R4Transformers.java [hunk 3]
```diff
@@ -16,6 +18,29 @@
...
+    }
+    return CodeableConcept.builder().coding(List.of(fhirCoding)).build();
+  }
+
+  /**
+   * Convert the optional coding to a FHIR coding and wrap it in a codeable concept. Returns null if
+   * it cannot be converted.
+   */
+  public static CodeableConcept asCodeableConceptWrapping(Optional<DatamartCoding> coding) {
+    if (coding.isEmpty()) {
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/R4Transformers.java:38]: `Transformers.isBlank()` here would also handle the case of an Optional that contains a blank string

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/procedure/R4ProcedureTransformer.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        return Status.not_done;
+      case completed:
+        return Status.completed;
+      default:
+        throw new IllegalArgumentException("Unsupported status: " + dmStatus);
+    }
+  }
+
+  static CodeableConcept statusReason(Optional<String> reasonNotPerformed) {
+    if (reasonNotPerformed.isEmpty()) {
```
Comment (bschofield-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/procedure/R4ProcedureTransformer.java:42]: `isBlank( ... )` to capture the empty value

**Pull request merged**